sklearn/metrics/_regression.py

```diff
@@ -887,15 +899,18 @@ def _assemble_r2_explained_variance(
             avg_weights = None
         elif multioutput == "variance_weighted":
             avg_weights = denominator
-            if not np.any(nonzero_denominator):
+            if not xp.any(nonzero_denominator):
                 # All weights are zero, np.average would raise a ZeroDiv error.
                 # This only happens when all y are constant (or 1-element long)
                 # Since weights are all equal, fall back to uniform weights.
                 avg_weights = None
     else:
         avg_weights = multioutput
 
-    return np.average(output_scores, weights=avg_weights)
+    result = _average(output_scores, weights=avg_weights)
+    if result.size == 1:
+        return float(result)
```

Comment:
`np.average` does not do that: it returns a scalar numpy array instead of a Python float scalar. Since we said that `_average` is partial Array API implementation of `np.average` I would rather try to mimic numpy instead of introducing original behaviors.

Reply:
I added a paragraph in 90076d3 to explain our conventions w.r.t. return types.

Hope that helps.